### ISSUE TITLE
Failure in `test_dataframe_merge` from `test_explicit_comms`

### ISSUE THREAD

**Issue by jakirkham** [rapidsai/dask-cuda#407]
When running the explicit comms test `test_dataframe_merge` locally with RAPIDS 0.16, am see the following error message.

```bash
$ python -m pytest -xvvv dask_cuda/tests/test_explicit_comms.py -k "test_dataframe_merge"
```

```python
______________________ test_dataframe_merge[tcp-pandas-1] ______________________

backend = 'pandas', protocol = 'tcp', nworkers = 1

    @pytest.mark.parametrize("nworkers", [1, 2, 4])
    @pytest.mark.parametrize("backend", ["pandas", "cudf"])
    @pytest.mark.parametrize("protocol", ["tcp", "ucx"])
    def test_dataframe_merge(backend, protocol, nworkers):
        p = mp.Process(target=_test_dataframe_merge, args=(backend, protocol, nworkers))
        p.start()
        p.join()
>       assert not p.exitcode
E       AssertionError: assert not 1
E        +  where 1 = <SpawnProcess name='SpawnProcess-1' pid=29098 parent=28991 stopped exitcode=1>.exitcode

dask_cuda/tests/test_explicit_comms.py:109: AssertionError
----------------------------- Captured stderr call -----------------------------
/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/site-packages/pandas/util/__init__.py:12: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing
distributed.worker - WARNING -  Compute Failed
Function:  _run_coroutine_on_worker
args:      (b"\xd1\xe2$'\x8b\x92L\x17\x8c\x98\x02\xf6\xcah\x06\xce", <function _dataframe_merge at 0x7f18ad4b2940>, ({0}, [{0: 2}, {0: 1}], [[('from_pandas-a80e8df8446f66803f12b385e7d6a4d5', 0), ('from_pandas-a80e8df8446f66803f12b385e7d6a4d5', 1)], [('from_pandas-4c4a905d6d6f41f93cdcfc515e26ce70', 0)]], ['key'], ['key']))
kwargs:    {}
Exception: AttributeError("'Series' object has no attribute 'merge'")

Process SpawnProcess-1:
Traceback (most recent call last):
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/datasets/jkirkham/devel/dask-cuda/dask_cuda/tests/test_explicit_comms.py", line 92, in _test_dataframe_merge
    ddf3 = dataframe_merge(ddf1, ddf2, on="key").set_index("key")
  File "/datasets/jkirkham/devel/dask-cuda/dask_cuda/explicit_comms/dataframe_merge.py", line 175, in dataframe_merge
    return comms.default_comms().dataframe_operation(
  File "/datasets/jkirkham/devel/dask-cuda/dask_cuda/explicit_comms/comms.py", line 240, in dataframe_operation
    return utils.dataframes_to_dask_dataframe(ret)
  File "/datasets/jkirkham/devel/dask-cuda/dask_cuda/explicit_comms/utils.py", line 48, in dataframes_to_dask_dataframe
    meta = c.submit(get_meta, dfs[0]).result()
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/site-packages/distributed/client.py", line 225, in result
    raise exc.with_traceback(tb)
  File "/datasets/jkirkham/devel/dask-cuda/dask_cuda/explicit_comms/comms.py", line 56, in _run_coroutine_on_worker
    return executor.submit(_run).result()
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/datasets/jkirkham/devel/dask-cuda/dask_cuda/explicit_comms/comms.py", line 53, in _run
    return future.result()
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/datasets/jkirkham/devel/dask-cuda/dask_cuda/explicit_comms/dataframe_merge.py", line 92, in _dataframe_merge
    return df1.merge(df2, left_on=left_on, right_on=right_on)
  File "/datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/site-packages/pandas/core/generic.py", line 5136, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'Series' object has no attribute 'merge'
=============================== warnings summary ===============================
../../miniconda/envs/rapids16dev/lib/python3.8/site-packages/pandas/util/__init__.py:12
  /datasets/jkirkham/miniconda/envs/rapids16dev/lib/python3.8/site-packages/pandas/util/__init__.py:12: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
    import pandas.util.testing

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED dask_cuda/tests/test_explicit_comms.py::test_dataframe_merge[tcp-pandas-1]
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
================= 1 failed, 14 deselected, 1 warning in 8.81s ==================
```

Note: While this failure case shows Pandas and TCP, we see all variants of this test failing (including cuDF and UCX cases).

Side note: The Pandas warning should be fixed with PR ( https://github.com/rapidsai/cudf/pull/6421 ).

**Comment by jakirkham** [comment]
Note this fails on `branch-0.16` currently and so is a release blocker.

cc @rjzamora @madsbk @pentschev @quasiben

**Comment by quasiben** [comment]
I can reproduce locally as well

**Comment by jakirkham** [comment]
So this failed for me when using the latest Dask + Distributed 2.29.0 release. However when I downgrade to Dask + Distributed 2.28.0, the error goes away. So suspect there is something in the recent release that broke this.

**Comment by quasiben** [comment]
Confirmed what @jakirkham pointed out.  Distributed 2.28 works for me

**Comment by jakirkham** [comment]
In an attempt to isolate this further, I mixed Dask 2.28.0 with Distributed 2.29.0 to see if it was coming from Dask or Distributed. It appears using Distributed 2.29.0 alone causes the issue.

**Comment by rjzamora** [comment]
> In an attempt to isolate this further, I mixed Dask 2.28.0 with Distributed 2.29.0 to see if it was coming from Dask or Distributed. It appears using Distributed 2.29.0 alone causes the issue.

Yes - Problem seems to be in distributed. I couldn't reproduce until I updated to latest.

**Comment by quasiben** [comment]
It looks like it originates here https://github.com/dask/distributed/pull/4127

At a968158416a790483a2a9a66a60fa270c2fdfa9e (just before PR 4127) the tests pass

**Comment by jakirkham** [comment]
Random guess, does PR ( https://github.com/dask/distributed/pull/4148 ) happen to fix it?

**Comment by quasiben** [comment]
Yup!  That fixes it!  Thanks @jakirkham @rjzamora !

**Comment by rjzamora** [comment]
> Yup! That fixes it!
I can confirm :)

**Comment by jakirkham** [comment]
Awesome! Thanks all 😄

**Comment by jrbourbeau** [comment]
We just released `distributed==2.30.0` which has @madsbk's fix in https://github.com/dask/distributed/pull/4148

**Comment by quasiben** [comment]
Thanks @jrbourbeau !
